src/protocol/ipa/mod.rs

```diff
@@ -962,12 +833,13 @@ pub mod tests {
 
             let world = TestWorld::new().await;
 
-            test_ipa_semi_honest(
+            test_ipa(
                 world,
                 &raw_data,
                 &expected_results,
                 per_user_cap,
                 MAX_BREAKDOWN_KEY,
+                false,
```

Comment:
it's a nit, but I really dislike boolean arguments that dictate which logic a subroutine must execute. it is very hard to read on the caller's site and understand what this `true` or `false` mean. imo it would be better if you could define an enum `IpaSecurityModel::{SemiHonest,Malicious}` (name is debatable ofc) and use it here.